pkg/codegen/convert/plugin_mapper.go

```diff
@@ -122,21 +172,21 @@ func (l *pluginMapper) GetMapping(provider string) ([]byte, error) {
 	for {
 		if len(l.plugins) == 0 {
 			// No plugins left to look in, return that we don't have a mapping
-			return nil, nil
+			return []byte{}, nil
 		}
 
 		// Pop the first spec off the plugins list
 		pluginSpec := l.plugins[0]
 		l.plugins = l.plugins[1:]
 
-		providerPlugin, err := l.host.Provider(pluginSpec.name, &pluginSpec.version)
+		providerPlugin, pluginCloser, err := l.providerFactory(pluginSpec.name, &pluginSpec.version)
 		if err != nil {
 			// We should maybe be lenient here and ignore errors but for now assume it's better to fail out on
 			// things like providers failing to start.
 			return nil, fmt.Errorf("could not create provider '%s': %w", pluginSpec.name, err)
 		}
 		defer func() {
-			contract.IgnoreError(l.host.CloseProvider(providerPlugin))
+			contract.IgnoreClose(pluginCloser)
```

Comment:
FYI, don't need `func() { .. }()` if using IgnoreClose because it won't run the actual close until afterwards.

Reply:
Ah yes good point, missed that little cleanup when changing this.